Documentation/ExtensionArchitecture/HowTo/ExtendExtbaseModel/Index.rst

```diff
@@ -0,0 +1,208 @@
+
+#.  :ref:`Extend the original model <extending-extbase-model_extend_original_model>`
+    in your custom extension or sitepackage.
+
+#.  Register your extended model with the according database table in
+    :file:`EXT:my_extension/Configuration/Extbase/Persistence/Classes.php`.
+
+#.  :ref:`Extend the original repository <extending-extbase-model_register_extended_model>`
+    in your custom extension or sitepackage.
+
+#.  :ref:`Register your extended repository <extending-extbase-model_register_extended_repository>`
+    to be used instead of the original one.
+
+Step by step
+============
+
+..  _extending-extbase-model_find_other_extending_models:
+
+Are you the only one trying to extend that model?
+-------------------------------------------------
+
+Within your installation you can search for other classes that are extending
+the original model or repository (see bellow).
+
+If the model is already extended but you only need to create the fields for
+your current installation you can proceed by extending the extended model.
+
+If the model has different types you can decide to introduce a new type and
+only extend that one type. This is for example commonly done when extending
+the model of EXT:news.
+
+If you are planing to publish your extension that extends another extensions
+model, research on `Packagist <https://packagist.org/>`__ and the
+`TER (TYPO3 extension repository) <https://extensions.typo3.org/>` for
+extensions that are already extending the model. If necessary but them in
+the `conflict` sections of you extensions :file:`composer.json` and
+:file:`ext_emconfig.php`.
+
+..  _extending-extbase-model_find_original_model:
+
+Find the original model
+-----------------------
+
+The model should be located in the original extension at path
+:file:`EXT:original_extension/Classes/Domain/Model` or a subdirectory
+thereof. If the model is not found here it might
+
+*   be situated in a different extension
+*   not be an Extbase model (you cannot follow this tutorial then)
+
+You can also try to debug the model in a Fluid template that outputs the model:
+
+..  code-block:: html
+    :caption: Some Fluid template that uses the model
+
+    <f:debug>{someModel}</f:debug>
+
+If you debugged the correct object the fully qualified PHP name of the model
+will appear in the debug output. This gives you further hints on where to find
+the associated class. You could for example do a full text search for the
+namespace of this class.
+
+If the class of the model is final:
+
+..  code-block:: php
+    :caption: EXT:original_extension/Classes/Domain/Model/SomeModel.php
+
+    final class SomeModel {
+        // ...
+    }
+
+It cannot be extended by means of this tutorial. Refer to the documentation of
+the original extension.
+
+..  _extending-extbase-model_find_original_repository:
+
+Find the original repository
+----------------------------
+
+In Extbase the repository of a model has to have the same class name as the
```

Comment:
Yesno, it can be reconfigured, but mostly is follows the convention.

Maybe "mostly has the same class name"

Reply:
Since when can this be reconfigured?

Reply:
In principle you can assign the custom type manually in your repository class:
https://github.com/TYPO3/typo3/blob/main/typo3/sysext/extbase/Classes/Persistence/Repository.php#L65

Reply:
Ok,  changed it